python/pyspark/sql/dataframe.py

```diff
@@ -351,8 +352,62 @@ def show(self, n=20, truncate=True, vertical=False):
         else:
             print(self._jdf.showString(n, int(truncate), vertical))
 
+    def _get_repl_config(self):
+        """Return the configs for eager evaluation each time when __repr__ or
+        _repr_html_ called by user or notebook.
+        """
+        eager_eval = self.sql_ctx.getConf(
+            "spark.sql.repl.eagerEval.enabled", "false").lower() == "true"
+        console_row = int(self.sql_ctx.getConf(
+            "spark.sql.repl.eagerEval.maxNumRows", u"20"))
+        console_truncate = int(self.sql_ctx.getConf(
+            "spark.sql.repl.eagerEval.truncate", u"20"))
+        return (eager_eval, console_row, console_truncate)
+
     def __repr__(self):
-        return "DataFrame[%s]" % (", ".join("%s: %s" % c for c in self.dtypes))
+        (eager_eval, console_row, console_truncate) = self._get_repl_config()
+        if not self._support_repr_html and eager_eval:
```

Comment:
What's `_support_repr_html` for?

Reply:
As comment before, this is the flag to check whether \_repr_html\_ is called.
![image](https://user-images.githubusercontent.com/4833765/40709259-2cbf6ede-6428-11e8-8cbe-e14e1450ec31.png)

Reply:
So you want to restrict `__repr__` to always return the original string like `"DataFrame[key: bigint, value: string]"` after `_repr_html_` is called?

Reply:
Yes that's right.

Reply:
I see, thanks.
I thinks it's okay, but I'm just curious why you want to restrict it?

Reply:
Just want to avoid calling `_jdf` twice here, cause the second time called by `__repr__ ` is useless while `_repr_html_` is supported. The return string of `__repr__` will not finally be shown to notebook.